graphjet-core/src/main/java/com/twitter/graphjet/algorithms/socialproof/SocialProofGenerator.java

```diff
@@ -173,10 +195,45 @@ private void collectRightNodeInfo(SocialProofRequest request) {
     }
   }
 
+  /**
+   * When the incoming request asks for Favorite as one of the social proofs, we must check for
+   * Unfavorite edges as well, and remove the corresponding unfavorited edges.
+   */
+  private boolean shouldRemoveUnfavoritedEdges(SocialProofRequest request) {
+    for (byte socialProofType: request.getSocialProofTypes()) {
+      if (socialProofType == FAVORITE_SOCIAL_PROOF_TYPE ||
+          socialProofType == UNFAVORITE_SOCIAL_PROOF_TYPE) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Given a {@link SocialProofRequest} object, create a new request that add Unfavorite as one of
+   * the social proof types to be collected.
+   */
+  private SocialProofRequest addUnfavoriteEdgeTypeToRequest(SocialProofRequest oldRequest) {
```

Comment:
Generally speaking, we should avoid creating request object inside an algorithm. The request is the input, and the underlying algorithm could parse it and convert it to something else internal, but rewriting requests is not a good idea.

In this case, do not you just need newSocialProofTypes?

Reply:
Agree, yes in this case we just need to find a way to squeeze in the unfavorite social proof types into the algorithm. 
I did it this way to decouple the unfavorite logic from the remainder of the class logic, though now thinking back I agree this is more of a hack than clean code. Let me refactor this.